core/types/state_account.go

```diff
@@ -27,109 +27,27 @@
-		Root:     EmptyRootHash,
-		CodeHash: EmptyCodeHash.Bytes(),
-	}
-}
-
-// Copy returns a deep-copied state account object.
-func (acct *StateAccount) Copy() *StateAccount {
-	var balance *uint256.Int
-	if acct.Balance != nil {
-		balance = new(uint256.Int).Set(acct.Balance)
-	}
-	return &StateAccount{
-		Nonce:       acct.Nonce,
-		Balance:     balance,
-		Root:        acct.Root,
-		CodeHash:    common.CopyBytes(acct.CodeHash),
-		IsMultiCoin: acct.IsMultiCoin,
-	}
-}
-
-// SlimAccount is a modified version of an Account, where the root is replaced
-// with a byte slice. This format can be used to represent full-consensus format
-// or slim format which replaces the empty root and code hash as nil byte slice.
-type SlimAccount struct {
-	Nonce       uint64
-	Balance     *uint256.Int
-	Root        []byte // Nil if root equals to types.EmptyRootHash
-	CodeHash    []byte // Nil if hash equals to types.EmptyCodeHash
-	IsMultiCoin bool
-}
+type (
+	// Import these types from the go-ethereum package
+	StateAccount = ethtypes.StateAccount
+	SlimAccount  = ethtypes.SlimAccount
+)
 
-// SlimAccountRLP encodes the state account in 'slim RLP' format.
-func SlimAccountRLP(account StateAccount) []byte {
-	slim := SlimAccount{
-		Nonce:       account.Nonce,
-		Balance:     account.Balance,
-		IsMultiCoin: account.IsMultiCoin,
-	}
-	if account.Root != EmptyRootHash {
-		slim.Root = account.Root[:]
-	}
-	if !bytes.Equal(account.CodeHash, EmptyCodeHash[:]) {
-		slim.CodeHash = account.CodeHash
-	}
-	data, err := rlp.EncodeToBytes(slim)
-	if err != nil {
-		panic(err)
-	}
-	return data
-}
+var (
+	// Import these functions from the go-ethereum package
+	NewEmptyStateAccount = ethtypes.NewEmptyStateAccount
+	SlimAccountRLP       = ethtypes.SlimAccountRLP
+	FullAccount          = ethtypes.FullAccount
+	FullAccountRLP       = ethtypes.FullAccountRLP
+)
 
-// FullAccount decodes the data on the 'slim RLP' format and returns
-// the consensus format account.
-func FullAccount(data []byte) (*StateAccount, error) {
-	var slim SlimAccount
-	if err := rlp.DecodeBytes(data, &slim); err != nil {
-		return nil, err
-	}
-	var account StateAccount
-	account.Nonce, account.Balance, account.IsMultiCoin = slim.Nonce, slim.Balance, slim.IsMultiCoin
+type isMultiCoin bool
 
-	// Interpret the storage root and code hash in slim format.
-	if len(slim.Root) == 0 {
-		account.Root = EmptyRootHash
-	} else {
-		account.Root = common.BytesToHash(slim.Root)
-	}
-	if len(slim.CodeHash) == 0 {
-		account.CodeHash = EmptyCodeHash[:]
-	} else {
-		account.CodeHash = slim.CodeHash
-	}
-	return &account, nil
-}
+var IsMultiCoinPayloads = ethtypes.RegisterExtras[isMultiCoin]()
 
-// FullAccountRLP converts data on the 'slim RLP' format into the full RLP-format.
-func FullAccountRLP(data []byte) ([]byte, error) {
-	account, err := FullAccount(data)
-	if err != nil {
-		return nil, err
-	}
-	return rlp.EncodeToBytes(account)
+func IsMultiCoin(a *StateAccount) bool {
```

Comment:
Can we also remove this if we don't want to support it in API?

Reply:
It's still used in a test (that verifies it round trips through the snapshot properly) so may be worth keeping